tests/rptest/tests/rbac_test.py

```diff
@@ -149,6 +172,110 @@ def test_invalid_create_role(self):
             with expect_http_error(400):
                 self.superuser_admin.create_role(role=invalid_rolename)
 
+    @cluster(num_nodes=1)
```

Comment:
I went with reducing the cluster size from 3 to 1 here to simplify the test. After the role creation commands on line 183, there's no guarantee that we'll see the new roles on all nodes.
The alternatives I saw are:
* Reduce cluster to 1 node -- I went with this as this seemed the simplest and lowest chance of being flaky
* Sleep for 5 seconds after role creation and assume that the roles propagate in that time reliably -- risky of being flaky
* Change `expect_roles_with_filter` to use `wait_until` -- this makes the code more complex, but I'm open to this as well

Reply:
[Force-pushed](https://github.com/redpanda-data/redpanda/compare/7fdf0422310188782f5799a554546154b4d79a52..669ac7b8ffe053bd57ec26aca28e0ed1bb6148f3) now after talking to @BenPope offline as he made me convinced that `wait_until` is the most reliable option here